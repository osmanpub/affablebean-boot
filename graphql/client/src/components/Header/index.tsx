import React from "react";
import { Link } from "react-router-dom";
import WidgetBar from "../WidgetBar";
import "./Header.css";
import { HeaderMain, HeaderWrapper, Logo } from "./Header.styles";

type Props = {
  url: string;
};

export default function Header(props: Props) {
  const { url } = props;

  return (
    <HeaderWrapper>
      <HeaderMain>
        <WidgetBar url={url} />
        <Logo>
          <Link to={"/"} data-cy="home-logo">
            <img src="/static/img/logo.png" alt="Affable Bean logo" />
          </Link>
        </Logo>
        <img
          alt="the affable bean"
          id="logoText"
          src="/static/img/logoText.png"
        />
      </HeaderMain>
    </HeaderWrapper>
  );
}
