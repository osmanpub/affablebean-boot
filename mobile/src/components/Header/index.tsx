import React from 'react';
import {
  Image,
  StyleSheet,
  Text,
  TouchableHighlight,
  TouchableWithoutFeedback,
  View,
} from 'react-native';
import {connect} from 'react-redux';
import {Cart} from '../../interfaces/cart';
import {RootState} from '../../redux';

type Props = {
  cart: Cart;
  currentScreen: string;
  setScreen: Function;
};

function Header(props: Props) {
  const {cart, currentScreen, setScreen} = props;

  let cartWidget = null;
  let checkoutWidget = null;

  if (cart && cart.numberOfItems > 0) {
    if (currentScreen !== 'Cart') {
      cartWidget = (
        <TouchableHighlight onPress={() => setScreen('Cart')}>
          <Text style={styles.checkout}>view cart</Text>
        </TouchableHighlight>
      );
    }

    if (currentScreen !== 'Checkout') {
      checkoutWidget = (
        <TouchableWithoutFeedback onPress={() => setScreen('Checkout')}>
          <Text style={styles.viewCart}>checkout</Text>
        </TouchableWithoutFeedback>
      );
    }
  }

  return (
    <View
      style={{
        alignItems: 'center',
      }}>
      <TouchableWithoutFeedback onPress={() => setScreen('Home')}>
        <Image source={require('./logo.jpg')} style={styles.logo} />
      </TouchableWithoutFeedback>

      {cart && cart.numberOfItems > 0 && (
        <View style={{flexDirection: 'row'}}>
          {cartWidget}
          {checkoutWidget}
          <View style={{flexDirection: 'row', padding: 8}}>
            <Image source={require('./cart.gif')} />
            <Text>&nbsp;{cart.numberOfItems}&nbsp;items</Text>
          </View>
        </View>
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  checkout: {
    borderColor: 'blue',
    borderRadius: 32,
    borderWidth: 1,
    color: 'blue',
    fontWeight: 'bold',
    padding: 8,
  },
  logo: {
    margin: 16,
  },
  viewCart: {
    borderColor: 'blue',
    borderRadius: 32,
    borderWidth: 1,
    color: 'blue',
    fontWeight: 'bold',
    marginBottom: 16,
    marginLeft: 16,
    marginRight: 16,
    padding: 8,
  },
});

const mapStateToProps = (state: RootState) => ({
  cart: state.cart,
});

export default connect(mapStateToProps)(Header);
