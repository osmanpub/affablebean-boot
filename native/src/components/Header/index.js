import React from "react";
import {
  Image,
  StyleSheet,
  Text,
  TouchableHighlight,
  View
} from "react-native";
import PropTypes from "prop-types";

export default function Header(props) {
  const { cart, currentScreen, setScreen } = props;

  const styles = StyleSheet.create({
    checkout: {
      borderColor: "blue",
      borderRadius: 32,
      borderWidth: 1,
      color: "blue",
      fontWeight: "bold",
      padding: 8
    },
    logo: {
      margin: 16
    },
    viewCart: {
      borderColor: "blue",
      borderRadius: 32,
      borderWidth: 1,
      color: "blue",
      fontWeight: "bold",
      marginBottom: 16,
      marginLeft: 16,
      marginRight: 16,
      padding: 8
    }
  });

  function goHome() {
    setScreen("Home");
  }

  function viewCart() {
    if (currentScreen !== "Cart") {
      setScreen("Cart");
    }
  }

  function viewCheckout() {
    if (currentScreen !== "Checkout") {
      setScreen("Checkout");
    }
  }

  return (
    <View
      style={{
        alignItems: "center",
        background: "#f7f7e9"
      }}
    >
      <TouchableHighlight onPress={goHome}>
        <Image source={require("./logo.jpg")} style={styles.logo} />
      </TouchableHighlight>

      {cart && cart.numberOfItems > 0 && (
        <View style={{ flexDirection: "row" }}>
          <TouchableHighlight onPress={viewCart}>
            <Text style={styles.checkout}>view cart</Text>
          </TouchableHighlight>
          <TouchableHighlight onPress={viewCheckout}>
            <Text style={styles.viewCart}>checkout</Text>
          </TouchableHighlight>
          <View style={{ flexDirection: "row", padding: 8 }}>
            <Image source={require("./cart.gif")} />
            <Text>&nbsp;{cart.numberOfItems}&nbsp;items</Text>
          </View>
        </View>
      )}
    </View>
  );
}

Header.propTypes = {
  cart: PropTypes.object,
  currentScreen: PropTypes.string.isRequired,
  setScreen: PropTypes.func.isRequired
};
