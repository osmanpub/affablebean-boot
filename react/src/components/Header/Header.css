#logoText {
  float: left;
  margin: 20px 0 0 70px;
  /* font styles apply to text within img alt attribute */
  font-family: "American Typewriter", Courier, monospace;
  font-size: 50px;
  color: #333;
}
