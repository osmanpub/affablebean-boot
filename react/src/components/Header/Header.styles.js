import styled from "styled-components";

export const HeaderWrapper = styled.div`
  background: #f7f7e9;
`;

export const HeaderMain = styled.div`
  width: 1080px;
  height: 250px;
  background: url(/static/img/stalk.png) no-repeat 28px 28px;
`;
