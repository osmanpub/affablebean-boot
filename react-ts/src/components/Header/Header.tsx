import React from "react";
import { Link } from "react-router-dom";
import { HeaderMain, HeaderWrapper, Logo } from "./Header.styles";
import WidgetBar from "../WidgetBar";
import "./Header.css";
import { HeaderProps } from "../../interfaces/props";

export function Header(props: HeaderProps) {
  const { cart, url } = props;

  return (
    <HeaderWrapper>
      <HeaderMain>
        <WidgetBar cart={cart} url={url} />
        <Logo>
          <Link to={"/"}>
            <img src="/static/img/logo.png" alt="Affable Bean logo" />
          </Link>
        </Logo>
        <img
          id="logoText"
          src="/static/img/logoText.png"
          alt="the affable bean"
        />
      </HeaderMain>
    </HeaderWrapper>
  );
}
