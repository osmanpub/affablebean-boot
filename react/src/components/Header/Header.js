import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { HeaderMain, HeaderWrapper, Logo } from "./Header.styles";
import WidgetBar from "../WidgetBar";
import "./Header.css";

export function Header(props) {
  const { cart, url } = props;

  return (
    <HeaderWrapper>
      <HeaderMain>
        <WidgetBar cart={cart} url={url} />
        <Logo>
          <Link to={"/"}>
            <img src="/static/img/logo.png" alt="Affable Bean logo" />
          </Link>
        </Logo>
        <img
          id="logoText"
          src="/static/img/logoText.png"
          alt="the affable bean"
        />
      </HeaderMain>
    </HeaderWrapper>
  );
}

Header.propTypes = {
  cart: PropTypes.object,
  url: PropTypes.string.isRequired
};
