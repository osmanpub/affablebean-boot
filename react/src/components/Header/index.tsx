import React from "react";
import { Link } from "react-router-dom";
import WidgetBar from "../WidgetBar";
import "./Header.css";
import { HeaderMain, HeaderWrapper, Logo } from "./Header.styles";

type Props = {
  url: string;
};

export default function Header(props: Props) {
  const { url } = props;

  return (
    <HeaderWrapper>
      <HeaderMain>
        <WidgetBar url={url} />
        <Logo>
          <Link to={"/"}>
            <img src="/static/img/logo.png" alt="Affable Bean logo" />
          </Link>
        </Logo>
        <img
          id="logoText"
          src="/static/img/logoText.png"
          alt="the affable bean"
        />
      </HeaderMain>
    </HeaderWrapper>
  );
}
