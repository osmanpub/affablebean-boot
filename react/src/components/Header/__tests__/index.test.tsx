import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import Header from "..";

const mockStore = configureStore([]);

const initialState = {
  cart: {
    didInvalidate: false,
    isFetching: false,
    items: [],
    numberOfItems: 0,
    subtotal: 0
  }
};

const store = mockStore(initialState);

describe("<Header />", () => {
  it("renders correctly", () => {
    const component = renderer
      .create(
        <Router>
          <Provider store={store}>
            <Header url="/" />
          </Provider>
          )
        </Router>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
