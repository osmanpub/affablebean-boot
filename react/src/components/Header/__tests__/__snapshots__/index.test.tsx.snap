// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
Array [
  <div
    className="sc-AxhCb kYRjhm"
  >
    <div
      className="sc-AxhUy EsjBI"
    >
      <div
        className="sc-AxjAm cHCNiJ"
      >
        <div
          className="sc-AxirZ enfYhw headerWidget"
        >
          <a
            className="bubble"
            data-cy="proceed-checkout"
            href="/checkout"
            onClick={[Function]}
          >
            proceed to checkout ➟
          </a>
        </div>
        <div
          className="sc-AxiKw fRiXyM headerWidget"
        >
          <span
            className="horizontalMargin"
            data-cy="cart-hdr-total"
            data-testid="cart-hdr-total"
          >
            <img
              alt="shopping cart icon"
              src="/static/img/cart.gif"
            />
             
            0 items
          </span>
          <a
            className="bubble"
            data-cy="cart-hdr-view-cart"
            href="/viewCart/false"
            onClick={[Function]}
          >
            view cart
          </a>
        </div>
      </div>
      <div
        className="sc-AxgMl kvztNL"
      >
        <a
          data-cy="home-logo"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="Affable Bean logo"
            src="/static/img/logo.png"
          />
        </a>
      </div>
      <img
        alt="the affable bean"
        id="logoText"
        src="/static/img/logoText.png"
      />
    </div>
  </div>,
  ")",
]
`;
